libsplinter/src/admin/rest_api/actix/circuits.rs

```diff
@@ -203,83 +206,95 @@ mod tests {
 
     #[test]
     /// Tests a GET /admin/circuits request with filter returns the expected circuit.
+    ///
+    /// This test binds to `127.0.0.1:8111`; this URL should not be bound to by other tests.
     fn test_list_circuit_with_filters_ok() {
-        let splinter_state = filled_splinter_state();
-
-        let mut app = test::init_service(
-            App::new().data(splinter_state.clone()).service(
-                web::resource("/admin/circuits")
-                    .route(web::get().to_async(list_circuits::<SplinterState>)),
-            ),
-        );
+        let bind_url = "127.0.0.1:8111";
```

Comment:
I wonder if we should bind to a random high port within a range.

Reply:
As in randomly generated ports? Seems like it could cause ephemeral failures (probably very infrequent, but still).

Reply:
Yeah, we would need some way to recover (then I suppose it doesn't need to be random).

Reply:
Updated to programmatically find an open port in a range